src/lib/HST/Util/Selectors.hs

```diff
@@ -89,3 +93,106 @@ getPatVarName (S.PTuple _ _ _) = reportFatal
 getPatVarName (S.PList _ _) = reportFatal
   $ Message Error
   $ "Expected variable or wildcard pattern, got list pattern."
+
+-------------------------------------------------------------------------------
+-- Find Identifiers in Modules                                               --
+-------------------------------------------------------------------------------
+-- Class for all types that contain identifiers.
+class HasIdentifiers a where
+  findIdentifiers :: a -> Set String
+
+instance HasIdentifiers a => HasIdentifiers [a] where
+  findIdentifiers = Set.unions . map findIdentifiers
+
+instance HasIdentifiers a => HasIdentifiers (Maybe a) where
+  findIdentifiers = maybe Set.empty findIdentifiers
```

Comment:
These two instances as well as the insatnces for `QName` and `Name`s don't have a Haddock comment.